cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/AnomalyDetectorState.java

```diff
@@ -287,12 +295,17 @@ private SelfHealingEnabledRatio selfHealingEnabledRatio() {
    * Update anomaly status once associated self-healing operation has finished.
    *
    * @param anomalyId Unique id of anomaly which triggered self-healing operation.
+   * @param completeWithError Whether the task execution finished with error.
    */
-  public synchronized void markSelfHealingFinished(String anomalyId) {
+  public synchronized void markSelfHealingFinished(String anomalyId, boolean completeWithError) {
     if (_ongoingSelfHealingAnomaly == null || !_ongoingSelfHealingAnomaly.anomalyId().equals(anomalyId)) {
       throw new IllegalStateException(String.format("Anomaly %s is not marked as %s state in AnomalyDetector.",
                                                     anomalyId, AnomalyState.Status.FIX_STARTED));
     }
+    long totalTime = _time.milliseconds() - _ongoingSelfHealingAnomaly.detectionTimeMs();
+    LOG.info("Self-healing for anomalyId {} completed{}. Total time: {} ms", anomalyId, completeWithError ? " with error" : "", totalTime);
+    // Time the duration no matter whether the self-healing is completed successfully or not.
+    _anomalyDetectToFixCompleteTimer.get(_ongoingSelfHealingAnomaly.anomalyType()).update(totalTime, TimeUnit.MILLISECONDS);
```

Comment:
I don't feel like this is a useful metric--how can we determine (from graphs) which ones succeeded and which failed? Separating those into two metrics would be much better since then we'd have both "time to fix the anomaly" and "time wasted attempting to heal but failing at it."

Reply:
Changed to only time the successful fix.  do not have a separate timer to track the failed fixes because the duration for failed fix is not very meaningful and there are other metrics/logs for tracking failures. There are two types of failures: (1) interrupted (receiving stop signal). There is already a counter to track the number of stopped execution. (2) encountering exception. This is unexpected and rare. We can use log if needed.

Reply:
Sounds good, thanks.